docs/it/refactoring.md

```diff
@@ -0,0 +1,136 @@
+L'attività di refactoring può essere paragonata a una pulizia profonda di casa, piuttosto che a una routine di sistemazione della propria cameretta, necessaria per mantenere il codice sicuro, di valore ed efficiente.
+
+## Quando è meglio evitare
+
+Non sempre è utile effettuare refactoring.
+Ci sono casi in cui il codice è già perfettamente ottimizzato anche se, a una vista inesperta, potrebbe risultare di difficile lettura.
+Ci sono casi in cui le performance sono fondamentali e i principi SOLID non sono sempre rispettati. In questo caso, il refactoring ci porterebbe a creare dei layer aggiuntivi di codice che, sebbene ci renderebbe il codice più comprensibile, avrebbe un impatto negativo sulle performance. Un esempio sono le librerie che permettono le connessioni con delle basi di dati, dove magari è fondamentale la gestione della latenza.
+In questi casi, ma più in generale, è necessario tenere a mente che l'attività di refactoring ha tra gli obiettivi quello di rivedere il codice rendendolo più mantenibile, nonché rifattorizzabile.
+Diretta conseguenza di questo obiettivo, è la sua applicazione nella pratica del TDD (Test Driven Development), dove il refactoring copre uno degli step.
+
+## Come
+
+Come si può intuire, con le attività di refactoring andremo a migliorare il codice in modo da renderlo meno complesso, più mantenibile, più resiliente.
+Per farlo, potremmo iniziare con:
+
+- Red-Green-Refactor;
+- Rinominare i metodi oscuri;
+- Cambiare la firma dei metodi;
+- Estrarre nuovi metodi;
+- Semplificare i metodi esistenti;
+- Astrarre.
+
+### Red-Green-Refactor
+
+Forse il metodo più popolare, utilizzato per mettere in pratica il TDD (Test Driven Development), dove chi sviluppa suddivide il processo di scrittura del codice in tre fasi:
+
+- Analizza quali sono le esigenze e scrive dapprima il test, che, per ovvi motivi, fallirà (_Red_);
+- Scrive il codice necessario affinché il test passi (_Green_);
+- Lo migliora (_Refactoring_)
+
+### Rinominare i metodi oscuri
+
+Ci sono metodi che hanno nomi degni di un codice morse ma che poco dicono sulle operazioni che vanno a intraprendere.
+Un buon inizio potrebbe essere quello di rinominarli in modo che chi legge abbia la possibilità di capire cosa fa quel metodo, senza necessità alcuna di andarlo a verificare.
+
+### Cambiare la firma dei metodi
+
+Partendo dal presupposto che il codice è vivo, si ci può trovare nelle condizioni in cui al metodo inizialmente venivano passate diverse informazioni, che magari ora non sono più necessarie. E magari non lo sono perché nel frattempo, quella parte di codice era stata già snellita in qualche attività precedente.
+Possiamo quindi rimuovere le informazioni in eccesso, modificandone la firma.
+Inoltre, nei linguaggi tipizzati, potremmo verificare se c'è la possibilità di rendere detta firma più stringente, ammettendo e/o restituendo tipi meno generici.
```

Comment:
Qui sottolineerei che è importante farlo solo nelle major updates, e che sarebbe preferibile utilizzare in prima battuta delle keywords o delle annotazioni speciali come `@deprecated` per comunicare a chi utilizza quei metodi che quei parametri non saranno più utilizzati a posteriori.

Reply:
Sì, ci sta. Apporto dei cambiamenti al periodo

Reply:
@mrkrash come sopra!